<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Synapter2 and synergise2 • synapter</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Synapter2 and synergise2">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">synapter</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">2.5.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/synapter.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/fragmentmatching.html">Fragment matching using *synapter*</a>
    </li>
    <li>
      <a href="../articles/synapter2.html">Synapter2 and synergise2</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Synapter2 and synergise2</h1>
                        <h4 class="author">Laurent Gatto</h4>
            <address class="author_afil">
      Computational Proteomics Unit, Cambridge, UK.<br><h4 class="author">Sebastian Gibb</h4>
            <address class="author_afil">
      Department of Anesthesiology and Intensive Care, University Medicine Greifswald, Germany.<br><h4 class="author">Pavel V. Shliaha</h4>
            <address class="author_afil">
      Department of Biochemistry and Molecular Biology, University of Southern Denmark, Denmark.<br><div class="hidden name"><code>synapter2.Rmd</code></div>

    </address>
</address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This vignette describes the new functionality implemented version 2.0 of the ‘synapter’ package. It describes the typical workflow including the 3D grid search, fragment matching and correction of detector saturation.</p>
    </div>
    
<div id="foreword" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#foreword" class="anchor"></a>Foreword</h1>
<p><a href="http://bioconductor.org/packages/synapter"><em>synapter</em></a> is free and open-source software. If you use it, please support the project by citing it in publications:</p>
<blockquote>
<p>Nicholas James Bond, Pavel Vyacheslavovich Shliaha, Kathryn S. Lilley, and Laurent Gatto. <em>Improving qualitative and quantitative performance for MS<span class="math inline">\(^E\)</span>-based label free proteomics.</em> J. Proteome Res., 2013, 12 (6), pp 2340–2353</p>
</blockquote>
</div>
<div id="questions-and-bugs" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#questions-and-bugs" class="anchor"></a>Questions and bugs</h1>
<p>For bugs, typos, suggestions or other questions, please file an issue in our tracking system (<a href="https://github.com/lgatto/synapter/issues" class="uri">https://github.com/lgatto/synapter/issues</a>) providing as much information as possible, a reproducible example and the output of <code>sessionInfo()</code>.</p>
<p>If you don’t have a GitHub account or wish to reach a broader audience for general questions about proteomics analysis using R, you may want to use the Bioconductor support site: <a href="https://support.bioconductor.org/" class="uri">https://support.bioconductor.org/</a>.</p>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Here we describe the new functionality implemented in <em>synapter 2.0</em>. Namely this vignette covers the utilisation of the new 3D grid search, the fragment matching, intensity modeling and correction of detector saturation.</p>
</div>
<div id="workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow" class="anchor"></a>Workflow</h1>
<p>The <em>synapter2</em> workflow is similar to the old one in <em>synapter1</em>. First it is necessary to use <em>PLGS</em> to create the <em>csv</em> (and <em>xml</em>) files. Therefore we refer the reader to the default <em><a href="http://bioconductor.org/packages/synapter">synapter</a></em> vignette, available <a href="https://bioconductor.org/packages/release/bioc/vignettes/synapter/inst/doc/synapter.html">online</a> and with <code><a href="../articles/synapter.html">vignette("synapter", package = "synapter")</a></code>.</p>
<p>In contrast to the original workflow the <code>final_fragment.csv</code> file for the identification run and a <code>Spectrum.xml</code> file for the quantification run are needed if the fragment matching should be applied.</p>
<p>Subsequently the original workflow is enhanced by the new <a href="#gridsearch">3D grid search</a> and the <a href="#intensitymodel">intensity modeling</a>. Afterwards the <a href="#fragmentmatching">fragment matching</a> could be applied. <em><a href="http://bioconductor.org/packages/MSnbase">MSnbase</a></em> <span class="citation">(Gatto and Lilley 2012)</span> is used for further analysis. The new <em><a href="http://bioconductor.org/packages/synapter">synapter</a></em> adds <a href="#synapterplgs"><em>synapter</em>/<em>PLGS</em> consensus filtering</a> and the <a href="#saturation">detector saturation correction</a> for <code>MSnSet</code>s.</p>
<div class="figure">
<img src="Figures/synapter2workflow.png" alt="New synapter workflow. Dark green boxes show the traditional synapter1 part and the light green boxes highlight the new synapter2 functionality."><p class="caption">New <em>synapter</em> workflow. Dark green boxes show the traditional <em>synapter1</em> part and the light green boxes highlight the new <em>synapter2</em> functionality.</p>
</div>
<div id="step-by-step-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#step-by-step-workflow" class="anchor"></a>Step-by-step workflow</h2>
<div id="create-a-synapter-object" class="section level3">
<h3 class="hasAnchor">
<a href="#create-a-synapter-object" class="anchor"></a>Create a <code>Synapter</code> object</h3>
<p>To demonstrate a typical step-by-step workflow we use example data that are available on <a href="http://proteome.sysbiol.cam.ac.uk/lgatto/synapter/data/" class="uri">http://proteome.sysbiol.cam.ac.uk/lgatto/synapter/data/</a>. There is also an <code>synobj2</code> object in <em><a href="http://bioconductor.org/packages/synapterdata">synapterdata</a></em> which contains the same data.</p>
<p>The <code>Synapter</code> constructor uses a named <code>list</code> of input files. Please note that we add <code>identfragments</code> (<code>final_fragment.csv</code>) and <code>quantspectra</code> (<code>Spectrum.xml</code>) because we want to apply the fragment matching later.</p>
<pre><code>## Please find the raw data at:
## http://proteome.sysbiol.cam.ac.uk/lgatto/synapter/data/
library("synapter")

inlist &lt;- list(
  identpeptide = "fermentor_03_sample_01_HDMSE_01_IA_final_peptide.csv.gz",
  identfragments = "fermentor_03_sample_01_HDMSE_01_IA_final_fragment.csv.gz",
  quantpeptide = "fermentor_02_sample_01_HDMSE_01_IA_final_peptide.csv.gz",
  quantpep3d = "fermentor_02_sample_01_HDMSE_01_Pep3DAMRT.csv.gz",
  quantspectra = "fermentor_02_sample_01_HDMSE_01_Pep3D_Spectrum.xml.gz",
  fasta = "S.cerevisiae_Uniprot_reference_canonical_18_03_14.fasta")

synobj2 &lt;- Synapter(inlist, master=FALSE)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">synobj2</code></pre></div>
<pre><code>## Object of class "Synapter" 
## Class version 2.0.0 
## Package version 1.99.0 
## Data files:
##  + Identification pep file: fermentor_03_sample_01_HDMSE_01_IA_final_peptide.csv.gz 
##  + Identification Fragment file: fermentor_02_sample_01_HDMSE_01_Pep3D_Spectrum.xml.gz 
##  + Quantitation pep file: fermentor_02_sample_01_HDMSE_01_IA_final_peptide.csv.gz 
##  + Quantitation Pep3DAMRT file: fermentor_02_sample_01_HDMSE_01_Pep3DAMRT.csv.gz 
##  + Quantitation Spectrum file: fermentor_02_sample_01_HDMSE_01_Pep3D_Spectrum.xml.gz 
##  + Fasta file: S.cerevisiae_Uniprot_reference_canonical_18_03_14.fasta 
## Log:
## [1] "Instance created on Tue Apr  4 09:01:34 2017"
## [2] "Read identification peptide data [11471,16]" 
## [ 10 lines ]
## [13] "Read identification fragment data [7078]"
## [14] "Read quantitation spectra [36539]"</code></pre>
</div>
<div id="filtering" class="section level3">
<h3 class="hasAnchor">
<a href="#filtering" class="anchor"></a>Filtering</h3>
<p>The first steps in each <em><a href="http://bioconductor.org/packages/synapter">synapter</a></em> analysis are filtering by peptide sequence, peptide length, ppm error and <em>false positive rate</em>.</p>
<p>Here we use the default values for each method. But the accompanying plotting methods should be used to find the best threshold:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Synapter.html">filterUniqueDbPeptides</a></span>(synobj2,
                       <span class="dt">missedCleavages=</span><span class="dv">0</span>,
                       <span class="dt">IisL=</span><span class="ot">TRUE</span>)
<span class="kw"><a href="../reference/Synapter.html">filterPeptideLength</a></span>(synobj2, <span class="dt">l=</span><span class="dv">7</span>)
<span class="kw"><a href="../reference/Synapter.html">plotFdr</a></span>(synobj2)</code></pre></div>
<p><img src="synapter2_files/figure-html/filtering-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Synapter.html">filterQuantPepScore</a></span>(synobj2, <span class="dt">method=</span><span class="st">"BH"</span>,
                    <span class="dt">fdr=</span><span class="fl">0.05</span>)
<span class="kw"><a href="../reference/Synapter.html">filterIdentPepScore</a></span>(synobj2, <span class="dt">method=</span><span class="st">"BH"</span>,
                    <span class="dt">fdr=</span><span class="fl">0.05</span>)
<span class="kw">par</span>(<span class="dt">mfcol=</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
<span class="kw"><a href="../reference/Synapter.html">plotPpmError</a></span>(synobj2, <span class="dt">what=</span><span class="st">"Ident"</span>)
<span class="kw"><a href="../reference/Synapter.html">plotPpmError</a></span>(synobj2, <span class="dt">what=</span><span class="st">"Quant"</span>)</code></pre></div>
<p><img src="synapter2_files/figure-html/filtering-2.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfcol=</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))
<span class="kw"><a href="../reference/Synapter.html">filterQuantPpmError</a></span>(synobj2, <span class="dt">ppm=</span><span class="dv">20</span>)
<span class="kw"><a href="../reference/Synapter.html">filterIdentPpmError</a></span>(synobj2, <span class="dt">ppm=</span><span class="dv">20</span>)
<span class="kw"><a href="../reference/Synapter.html">plotPepScores</a></span>(synobj2)</code></pre></div>
<p><img src="synapter2_files/figure-html/filtering-3.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Synapter.html">filterIdentProtFpr</a></span>(synobj2, <span class="dt">fpr=</span><span class="fl">0.05</span>)
<span class="kw"><a href="../reference/Synapter.html">filterQuantProtFpr</a></span>(synobj2, <span class="dt">fpr=</span><span class="fl">0.05</span>)</code></pre></div>
</div>
<div id="modeling-retention-time" class="section level3">
<h3 class="hasAnchor">
<a href="#modeling-retention-time" class="anchor"></a>Modeling retention time</h3>
<p>Next we merge the identified peptides from the identification run and quantification run and build a <a href="https://en.wikipedia.org/wiki/Local_regression"><em>LOWESS</em></a> based retention time model to remove systematic shifts in the retention times. Here we use the default values but as stated above the plotting methods should be used to find sensible thresholds.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Synapter.html">mergePeptides</a></span>(synobj2)

<span class="kw"><a href="../reference/Synapter.html">plotRt</a></span>(synobj2, <span class="dt">what=</span><span class="st">"data"</span>)</code></pre></div>
<p><img src="synapter2_files/figure-html/rtmodel-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Synapter.html">setLowessSpan</a></span>(synobj2, <span class="dt">span=</span><span class="fl">0.05</span>)
<span class="kw"><a href="../reference/Synapter.html">modelRt</a></span>(synobj2)

<span class="kw">par</span>(<span class="dt">mfcol=</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
<span class="kw"><a href="../reference/Synapter.html">plotRtDiffs</a></span>(synobj2)
<span class="kw"><a href="../reference/Synapter.html">plotRt</a></span>(synobj2, <span class="dt">what=</span><span class="st">"model"</span>, <span class="dt">nsd=</span><span class="dv">1</span>)</code></pre></div>
<p><img src="synapter2_files/figure-html/rtmodel-2.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfcol=</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))

<span class="kw"><a href="../reference/Synapter.html">plotFeatures</a></span>(synobj2, <span class="dt">what=</span><span class="st">"all"</span>, <span class="dt">ionmobility=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><img src="synapter2_files/figure-html/rtmodel-3.png" width="700"></p>
</div>
<div id="gridsearch" class="section level3">
<h3 class="hasAnchor">
<a href="#gridsearch" class="anchor"></a>Grid search</h3>
<p>To find <em>EMRTS</em> (exact <em>m/z</em>-retention time pairs) we try are running a grid search to find the best retention time tolerance and <em>m/z</em> tolerance that results in the most correct one-to-one matching in the merged (already identified) data. If the identification and quantitation run are HDMS<span class="math inline">\(^E\)</span> data we could use the new 3D grid search that looks for the best matching in the retention time, <em>m/z</em> and ion mobility (drift time) domain to increase the accuracy. If one or both datasets are MS<span class="math inline">\(^E\)</span> data it falls back to the traditional 2D grid search.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Synapter.html">searchGrid</a></span>(synobj2,
           <span class="dt">imdiffs=</span><span class="kw">seq</span>(<span class="dt">from=</span><span class="fl">0.6</span>, <span class="dt">to=</span><span class="fl">1.6</span>, <span class="dt">by=</span><span class="fl">0.2</span>),
           <span class="dt">ppms=</span><span class="kw">seq</span>(<span class="dt">from=</span><span class="dv">2</span>, <span class="dt">to=</span><span class="dv">20</span>, <span class="dt">by=</span><span class="dv">2</span>),
           <span class="dt">nsds=</span><span class="kw">seq</span>(<span class="dt">from=</span><span class="fl">0.5</span>, <span class="dt">to=</span><span class="dv">5</span>, <span class="dt">by=</span><span class="fl">0.5</span>))
<span class="kw"><a href="../reference/Synapter.html">setBestGridParams</a></span>(synobj2)
<span class="kw"><a href="../reference/Synapter.html">findEMRTs</a></span>(synobj2)
<span class="kw"><a href="../reference/Synapter.html">plotEMRTtable</a></span>(synobj2)</code></pre></div>
<p><img src="synapter2_files/figure-html/gridsearch-1.png" width="700"></p>
</div>
<div id="fragmentmatching" class="section level3">
<h3 class="hasAnchor">
<a href="#fragmentmatching" class="anchor"></a>Fragment matching</h3>
<p>For the details of the fragment matching procedure we refer to the fragment matching vignette that is available <a href="https://bioconductor.org/packages/release/bioc/vignettes/synapter/inst/doc/fragmentmatching.html">online</a> and with <code><a href="../articles/fragmentmatching.html">vignette("fragmentmatching", package = "synapter")</a></code>. Briefly we compare the fragments of the identification run with the spectra from the quantification run and remove entries where there are very few/none common peaks/fragments between them.</p>
<p>First we starting by removing less intense fragments and peaks.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Synapter.html">filterFragments</a></span>(synobj2,
                <span class="dt">what=</span><span class="st">"fragments.ident"</span>,
                <span class="dt">minIntensity=</span><span class="dv">70</span>)
<span class="kw"><a href="../reference/Synapter.html">filterFragments</a></span>(synobj2,
                <span class="dt">what=</span><span class="st">"spectra.quant"</span>,
                <span class="dt">minIntensity=</span><span class="dv">70</span>)</code></pre></div>
<p>Next we look for common peaks via <code>fragmentMatching</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Synapter.html">fragmentMatching</a></span>(synobj2)</code></pre></div>
<pre><code>## Warning in object$fragmentMatching(verbose = verbose): FragmentMatching ppm
## tolerance undefined. Setting to default value.</code></pre>
<p>We get tables for unique and non-unique matches:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw"><a href="../reference/Synapter.html">fragmentMatchingPerformance</a></span>(synobj2,
                                         <span class="dt">what=</span><span class="st">"unique"</span>))</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="right">ncommon</th>
<th align="right">tp</th>
<th align="right">fp</th>
<th align="right">tn</th>
<th align="right">fn</th>
<th align="right">all</th>
<th align="right">fdr</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">0</td>
<td align="right">3097</td>
<td align="right">79</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4159</td>
<td align="right">0.0248741</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">2556</td>
<td align="right">22</td>
<td align="right">57</td>
<td align="right">541</td>
<td align="right">2864</td>
<td align="right">0.0085337</td>
</tr>
<tr class="odd">
<td align="right">2</td>
<td align="right">1682</td>
<td align="right">12</td>
<td align="right">67</td>
<td align="right">1415</td>
<td align="right">1864</td>
<td align="right">0.0070838</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">1193</td>
<td align="right">4</td>
<td align="right">75</td>
<td align="right">1904</td>
<td align="right">1339</td>
<td align="right">0.0033417</td>
</tr>
<tr class="odd">
<td align="right">4</td>
<td align="right">862</td>
<td align="right">1</td>
<td align="right">78</td>
<td align="right">2235</td>
<td align="right">987</td>
<td align="right">0.0011587</td>
</tr>
<tr class="even">
<td align="right">5</td>
<td align="right">651</td>
<td align="right">1</td>
<td align="right">78</td>
<td align="right">2446</td>
<td align="right">754</td>
<td align="right">0.0015337</td>
</tr>
<tr class="odd">
<td align="right">6</td>
<td align="right">488</td>
<td align="right">1</td>
<td align="right">78</td>
<td align="right">2609</td>
<td align="right">578</td>
<td align="right">0.0020450</td>
</tr>
<tr class="even">
<td align="right">7</td>
<td align="right">372</td>
<td align="right">1</td>
<td align="right">78</td>
<td align="right">2725</td>
<td align="right">453</td>
<td align="right">0.0026810</td>
</tr>
<tr class="odd">
<td align="right">8</td>
<td align="right">294</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">2803</td>
<td align="right">367</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">9</td>
<td align="right">240</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">2857</td>
<td align="right">307</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">10</td>
<td align="right">186</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">2911</td>
<td align="right">245</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">11</td>
<td align="right">149</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">2948</td>
<td align="right">197</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">12</td>
<td align="right">124</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">2973</td>
<td align="right">165</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">13</td>
<td align="right">95</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3002</td>
<td align="right">126</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">14</td>
<td align="right">82</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3015</td>
<td align="right">105</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">15</td>
<td align="right">59</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3038</td>
<td align="right">76</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">16</td>
<td align="right">48</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3049</td>
<td align="right">61</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">17</td>
<td align="right">41</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3056</td>
<td align="right">53</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">18</td>
<td align="right">27</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3070</td>
<td align="right">36</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">19</td>
<td align="right">21</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3076</td>
<td align="right">26</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">20</td>
<td align="right">14</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3083</td>
<td align="right">16</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">21</td>
<td align="right">12</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3085</td>
<td align="right">14</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">22</td>
<td align="right">8</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3089</td>
<td align="right">10</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">23</td>
<td align="right">6</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3091</td>
<td align="right">8</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">24</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3092</td>
<td align="right">6</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">25</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3093</td>
<td align="right">5</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">26</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3094</td>
<td align="right">3</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">27</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3094</td>
<td align="right">3</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">28</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">79</td>
<td align="right">3096</td>
<td align="right">1</td>
<td align="right">0.0000000</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw"><a href="../reference/Synapter.html">fragmentMatchingPerformance</a></span>(synobj2,
                                         <span class="dt">what=</span><span class="st">"non-unique"</span>))</code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="right">deltacommon</th>
<th align="right">tp</th>
<th align="right">fp</th>
<th align="right">tn</th>
<th align="right">fn</th>
<th align="right">all</th>
<th align="right">fdr</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">0</td>
<td align="right">163</td>
<td align="right">33</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4159</td>
<td align="right">0.1683673</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">163</td>
<td align="right">2</td>
<td align="right">31</td>
<td align="right">0</td>
<td align="right">374</td>
<td align="right">0.0121212</td>
</tr>
<tr class="odd">
<td align="right">2</td>
<td align="right">132</td>
<td align="right">1</td>
<td align="right">32</td>
<td align="right">31</td>
<td align="right">304</td>
<td align="right">0.0075188</td>
</tr>
<tr class="even">
<td align="right">3</td>
<td align="right">119</td>
<td align="right">1</td>
<td align="right">32</td>
<td align="right">44</td>
<td align="right">276</td>
<td align="right">0.0083333</td>
</tr>
<tr class="odd">
<td align="right">4</td>
<td align="right">110</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">53</td>
<td align="right">256</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">5</td>
<td align="right">95</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">68</td>
<td align="right">226</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">6</td>
<td align="right">85</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">78</td>
<td align="right">202</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">7</td>
<td align="right">75</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">88</td>
<td align="right">178</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">8</td>
<td align="right">67</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">96</td>
<td align="right">161</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">9</td>
<td align="right">60</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">103</td>
<td align="right">145</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">10</td>
<td align="right">53</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">110</td>
<td align="right">127</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">11</td>
<td align="right">43</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">120</td>
<td align="right">106</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">12</td>
<td align="right">37</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">126</td>
<td align="right">92</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">13</td>
<td align="right">27</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">136</td>
<td align="right">71</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">14</td>
<td align="right">20</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">143</td>
<td align="right">56</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">15</td>
<td align="right">12</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">151</td>
<td align="right">37</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">16</td>
<td align="right">10</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">153</td>
<td align="right">26</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">17</td>
<td align="right">8</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">155</td>
<td align="right">21</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">18</td>
<td align="right">7</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">156</td>
<td align="right">18</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">19</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">159</td>
<td align="right">9</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">20</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">161</td>
<td align="right">5</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">21</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">161</td>
<td align="right">5</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">22</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">161</td>
<td align="right">5</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">23</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">162</td>
<td align="right">3</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="right">24</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">162</td>
<td align="right">3</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="right">25</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">33</td>
<td align="right">162</td>
<td align="right">3</td>
<td align="right">0.0000000</td>
</tr>
</tbody>
</table>
<p>Subsequently we could filter by minimal accepted common peaks:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Synapter.html">filterUniqueMatches</a></span>(synobj2, <span class="dt">minNumber=</span><span class="dv">1</span>)
<span class="kw"><a href="../reference/Synapter.html">filterNonUniqueMatches</a></span>(synobj2, <span class="dt">minDelta=</span><span class="dv">2</span>)
<span class="kw"><a href="../reference/Synapter.html">filterNonUniqueIdentMatches</a></span>(synobj2)</code></pre></div>
<p>Finally we rescue EMRTs that are filtered but were identified by <em>PLGS</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Synapter.html">rescueEMRTs</a></span>(synobj2, <span class="dt">method=</span><span class="st">"rescue"</span>)</code></pre></div>
</div>
<div id="intensitymodel" class="section level3">
<h3 class="hasAnchor">
<a href="#intensitymodel" class="anchor"></a>Modeling intensity</h3>
<p>In a similar manner as correcting for the retention time drift we correct systematic errors of the intensity via a <a href="https://en.wikipedia.org/wiki/Local_regression"><em>LOWESS</em></a> model. The function <code>modelIntensity</code> has to applied after <code>findEMRTs</code>. The model is build on the merged peptides as it is done for the retention time model. But in contrast to the retention time model the prediction is necessary for the matched quantitation data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Synapter.html">plotIntensity</a></span>(synobj2, <span class="dt">what=</span><span class="st">"data"</span>)</code></pre></div>
<p><img src="synapter2_files/figure-html/intmodel-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/Synapter.html">setLowessSpan</a></span>(synobj2, <span class="fl">0.05</span>)
<span class="kw"><a href="../reference/Synapter.html">modelIntensity</a></span>(synobj2)
<span class="kw"><a href="../reference/Synapter.html">plotIntensity</a></span>(synobj2, <span class="dt">what=</span><span class="st">"model"</span>, <span class="dt">nsd=</span><span class="dv">1</span>)</code></pre></div>
<p><img src="synapter2_files/figure-html/intmodel-2.png" width="700"></p>
</div>
</div>
<div id="synergise2" class="section level2">
<h2 class="hasAnchor">
<a href="#synergise2" class="anchor"></a><code>synergise2</code>
</h2>
<p>The whole workflow described in the <a href="#stepbystep">step-by-step workflow</a> is wrapped in the <code>synergise2</code> function. As side effect it generates a nice <em>HTML</em> report. An example could be found on <a href="https://github.com/lgatto/synapter" class="uri">https://github.com/lgatto/synapter</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">synobj2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/synergise2.html">synergise2</a></span>(<span class="dt">filenames =</span> inlist,
                      <span class="dt">outputdir =</span> <span class="st">"."</span>)</code></pre></div>
</div>
</div>
<div id="synapterplgs" class="section level1">
<h1 class="hasAnchor">
<a href="#synapterplgs" class="anchor"></a><em>synapter</em>/<em>PLGS</em> agreement</h1>
<p>For the next steps we need to convert the <code>Synapter</code> object into an <code>MSnSet</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msn &lt;-<span class="st"> </span><span class="kw">as</span>(synobj2, <span class="st">"MSnSet"</span>)</code></pre></div>
<p>Subsequently we look for <em>synapter</em>/<em>PLGS</em> agreement (this is more useful for a combined <code>MSnSet</code>; see basic <em><a href="http://bioconductor.org/packages/synapter">synapter</a></em> <a href="https://bioconductor.org/packages/release/bioc/vignettes/synapter/inst/doc/synapter.html#combine">online</a> or with <code><a href="../articles/synapter.html">vignette("synapter", package = "synapter")</a></code>). <code>synapterPlgsAgreement</code> adds an agreement column for each sample and counts the agreement/disagreement in additional columns:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msn &lt;-<span class="st"> </span><span class="kw"><a href="../reference/synapterPlgsAgreement.html">synapterPlgsAgreement</a></span>(msn)
knitr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/knitr/topics/kable">kable</a></span>(<span class="kw">head</span>(<span class="kw">fData</span>(msn)[, <span class="kw">grepl</span>(<span class="st">"[Aa]gree"</span>,
                                     <span class="kw">fvarLabels</span>(msn)),
                             <span class="dt">drop=</span><span class="ot">FALSE</span>]))</code></pre></div>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="left">synapterPlgsAgreement.fermentor_02_sample_01_HDMSE_01</th>
<th align="right">nAgree</th>
<th align="right">nDisagree</th>
<th align="right">synapterPlgsAgreementRatio</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>NVNDVIAPAFVK</td>
<td align="left">agree</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td>TAGIQIVADDLTVTNPK</td>
<td align="left">agree</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td>WLTGPQLADLYHSLMK</td>
<td align="left">agree</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td>AAQDSFAAGWGVMVSHR</td>
<td align="left">agree</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td>TFAEALR</td>
<td align="left">agree</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td>LGANAILGVSLAASR</td>
<td align="left">no_synapter_transfer</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">NaN</td>
</tr>
</tbody>
</table>
</div>
<div id="saturation" class="section level1">
<h1 class="hasAnchor">
<a href="#saturation" class="anchor"></a>Correction of detector saturation</h1>
<p>As described in <span class="citation">(Shliaha et al. 2013)</span> Synapt G2 devices suffer from detector saturation. This could be partly corrected by <code>requantify</code>. Therefore a <code>saturationThreshold</code> has to be given above that intensity saturation potentially happens. There are several methods available.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msncor &lt;-<span class="st"> </span><span class="kw"><a href="../reference/requantify.html">requantify</a></span>(msn,
                     <span class="dt">saturationThreshold=</span><span class="fl">1e5</span>,
                     <span class="dt">method=</span><span class="st">"sum"</span>)</code></pre></div>
<p>If an <code>MSnSet</code> object was requantified using the <code>"sum"</code> requantification method <em>TOP3</em> normalisation is not valid anymore because the most abundant proteins are penalised by removing high intensity isotopes (for details see <code><a href="../reference/requantify.html">?requantify</a></code> and <code><a href="../reference/rescaleForTop3.html">?rescaleForTop3</a></code>). This could be overcome by calling <code>rescaleForTop3</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">msncor &lt;-<span class="st"> </span><span class="kw"><a href="../reference/rescaleForTop3.html">rescaleForTop3</a></span>(<span class="dt">before=</span>msn,
                         <span class="dt">after=</span>msncor,
                         <span class="dt">saturationThreshold=</span><span class="fl">1e5</span>)</code></pre></div>
</div>
<div id="new-functions-not-covered-in-this-vignette" class="section level1">
<h1 class="hasAnchor">
<a href="#new-functions-not-covered-in-this-vignette" class="anchor"></a>New functions not covered in this vignette</h1>
<p>Since <em><a href="http://bioconductor.org/packages/synapter">synapter</a></em> 2.0 <code>makeMaster</code> supports fragment files as well. It is possible to create a fragment library that could used for <a href="#fragmentmatching">fragment matching</a> because of the large data this could not covered in this vignette. An introduction how to create a <em>master</em> could be found in the basic <em><a href="http://bioconductor.org/packages/synapter">synapter</a></em> vignette, available <a href="https://bioconductor.org/packages/release/bioc/vignettes/synapter/inst/doc/synapter.html">online</a> or with <code><a href="../articles/synapter.html">vignette("synapter", package = "synapter")</a></code>. Please find details about creating a fragment library in <code><a href="../reference/makeMaster.html">?makeMaster</a></code>.</p>
</div>
<div id="sec:sessionInfo" class="section level1">
<h1 class="hasAnchor">
<a href="#sec:sessionInfo" class="anchor"></a>Session information</h1>
<p>All software and respective versions used to produce this document are listed below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.5.1 Patched (2018-08-16 r75161)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 14.04.5 LTS
## 
## Matrix products: default
## BLAS: /usr/lib/atlas-base/atlas/libblas.so.3.0
## LAPACK: /usr/lib/lapack/liblapack.so.3.0
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] synapterdata_1.18.0 synapter_2.5.2      MSnbase_2.7.4      
##  [4] ProtGenerics_1.12.0 BiocParallel_1.14.2 mzR_2.15.2         
##  [7] Rcpp_0.12.18        Biobase_2.40.0      BiocGenerics_0.26.0
## [10] BiocStyle_2.8.2    
## 
## loaded via a namespace (and not attached):
##  [1] vsn_3.48.1            splines_3.5.1         foreach_1.4.4        
##  [4] assertthat_0.2.0      highr_0.7             affy_1.58.0          
##  [7] stats4_3.5.1          yaml_2.2.0            impute_1.54.0        
## [10] pillar_1.3.0          backports_1.1.2       lattice_0.20-35      
## [13] glue_1.3.0            limma_3.36.3          digest_0.6.16        
## [16] RColorBrewer_1.1-2    XVector_0.20.0        qvalue_2.12.0        
## [19] colorspace_1.3-2      htmltools_0.3.6       preprocessCore_1.42.0
## [22] Matrix_1.2-14         plyr_1.8.4            MALDIquant_1.18      
## [25] XML_3.98-1.16         pkgconfig_2.0.2       bookdown_0.7         
## [28] zlibbioc_1.26.0       purrr_0.2.5           scales_1.0.0         
## [31] affyio_1.50.0         cleaver_1.18.0        tibble_1.4.2         
## [34] IRanges_2.14.11       ggplot2_3.0.0         lazyeval_0.2.1       
## [37] survival_2.42-6       magrittr_1.5          crayon_1.3.4         
## [40] memoise_1.1.0         evaluate_0.11         fs_1.2.6             
## [43] doParallel_1.0.11     MASS_7.3-50           xml2_1.2.0           
## [46] BiocInstaller_1.30.0  tools_3.5.1           hms_0.4.2            
## [49] stringr_1.3.1         S4Vectors_0.18.3      munsell_0.5.0        
## [52] bindrcpp_0.2.2        Biostrings_2.48.0     pcaMethods_1.72.0    
## [55] compiler_3.5.1        pkgdown_1.1.0         mzID_1.18.0          
## [58] rlang_0.2.2           grid_3.5.1            iterators_1.0.10     
## [61] rmarkdown_1.10        gtable_0.2.0          codetools_0.2-15     
## [64] multtest_2.36.0       roxygen2_6.1.0        reshape2_1.4.3       
## [67] R6_2.2.2              knitr_1.20            dplyr_0.7.6          
## [70] bindr_0.1.1           commonmark_1.5        rprojroot_1.3-2      
## [73] readr_1.1.1           desc_1.2.0            stringi_1.2.4        
## [76] tidyselect_0.2.4      xfun_0.3</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Gatto2012">
<p>Gatto, Laurent, and Kathryn S Lilley. 2012. “MSnbase – an R/Bioconductor Package for Isobaric Tagged Mass Spectrometry Data Visualization, Processing and Quantitation.” <em>Bioinformatics</em> 28 (2): 288–9. doi:<a href="https://doi.org/10.1093/bioinformatics/btr645">10.1093/bioinformatics/btr645</a>.</p>
</div>
<div id="ref-Shliaha2013">
<p>Shliaha, Pavel Vyacheslavovich, Nicholas James Bond, Laurent Gatto, and Kathryn S. Lilley. 2013. “The Effects of Travelling Wave Ion Mobility Separation on Data Independent Acquisition in Proteomics Studies.” <em>Journal of Proteome Research</em> 2 (6). doi:<a href="https://doi.org/10.1021/pr300775k">10.1021/pr300775k</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#foreword">Foreword</a></li>
      <li><a href="#questions-and-bugs">Questions and bugs</a></li>
      <li><a href="#introduction">Introduction</a></li>
      <li>
<a href="#workflow">Workflow</a><ul class="nav nav-pills nav-stacked">
<li><a href="#step-by-step-workflow">Step-by-step workflow</a></li>
      <li><a href="#synergise2"><code>synergise2</code></a></li>
      </ul>
</li>
      <li><a href="#synapterplgs"><em>synapter</em>/<em>PLGS</em> agreement</a></li>
      <li><a href="#saturation">Correction of detector saturation</a></li>
      <li><a href="#new-functions-not-covered-in-this-vignette">New functions not covered in this vignette</a></li>
      <li><a href="#sec:sessionInfo">Session information</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Laurent Gatto, Nick J. Bond, Pavel V. Shliaha, Sebastian Gibb.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
